### ISSUE TITLE
Sending AdaptiveCard v1.1 cause ServiceError on Facebook channel

### ISSUE THREAD

**Issue by yochail** [Microsoft/botbuilder-dotnet#1557]
## Version
What package version of the SDK are you using.
Bot.Builder 4.3.2
Bot.Connector 4.3.2
AdaptiveCards 1.1.2

## Describe the bug
When sending an adapter card using version 1.1 in the emulator - everything works great.
When trying the same with facebook channel, the connector is returning 500 server exception with: "code": "ServiceError"

## To Reproduce
send message request to connector:
POST https://facebook.botframework.com/v3/conversations/<con-ID>/activities/<act-ID>
data:
```
{
	"type": "message",
	"timestamp": "2019-03-26T10:06:11.8086503+00:00",
	"serviceUrl": "https://facebook.botframework.com/",
	"channelId": "facebook",
	"from": {
		"id": "<ID>",
		"name": "say2eat-test"
	},
	"conversation": {
		"isGroup": false,
		"id": "<ID>"
	},
	"recipient": {
		"id": "<ID>",
		"name": "Bou Tiki"
	},
	"text": "sdsdsd",

"attachments": [{
		"contentType": "application/vnd.microsoft.card.adaptive",
		"content": {
			"type": "AdaptiveCard",
			"version": "1.1", //***********this will not work. 1.0 will work****************
			"body": [] //some adaptive cards body or even empty
		}
	},
],
	"replyToId": "2qtspChXQVAvKQw6RNdMIhqYZQwOpNAYm79NJscsZRvBCgowFTF8E8UgA76pZPKY5zWN7B7RyMdiXe3KlhxJYQ"
}
```

## Expected behavior
the message will be forward to fb channel

[bug]

**Comment by v-kydela** [comment]
@yochail - Most channels only support Adaptive Cards 1.0. Are there any features of Adaptive Cards 1.1 your bot is using?

**Comment by yochail** [comment]
not specifically, but if this is the situation I couldn't find this in the documentation, and it would be nice to get an explanatory error message

**Comment by v-kydela** [comment]
@benbrown 
@CoHealer 

I don't know that there's anything the support team can do here. Beyond suggesting that Adaptive Cards 1.0 should be used, the only resolutions I can see are:

1. Update the Bot Connector to support Adaptive Cards 1.1 in the Facebook channel
2. Provide a more helpful error message when an activity fails because of Adaptive Cards 1.1

**Comment by v-kydela** [comment]
I emailed some team members about the issue a few days ago, but no plan of action has been decided on.

**Comment by CoHealer** [comment]
@v-kydela, can you confirm that this issue is specific to Facebook and does not repro in other supported 3rd party channels?

**Comment by v-kydela** [comment]
@CoHealer - This issue surely repros in most channels and is not specific to Facebook. Based on your question I am currently compiling data regarding which channels support which versions of Adaptive Cards.

**Comment by v-kydela** [comment]
@CoHealer - All right, I tested GroupMe, Kik, Slack, Telegram, Twilio, and Facebook. The only other channel that returns a ServiceError is Telegram, but that channel also returns BadRequest for Adaptive Cards 1.0. The other channels all fail without error when sent an Adaptive Card 1.1.

I hadn't realized this before, but there aren't any 3rd-party channels that actually support Adaptive Cards. Except for Telegram, they all just render the card as an image.

@yochail - Given that Facebook doesn't support Adaptive Cards, I'm changing my recommendation. You may find better functionality by using the Messenger platform's native template system. This blog post can assist you: https://blog.botframework.com/2017/03/28/custom-channel-data/

**Comment by CoHealer** [comment]
Closing as resolved

**Comment by artemious7** [comment]
Something similar happened to my bot. It's connected only to Telegram channel, and it worked fine until some time ago, when it started to fail to send my Adaptive Cards. Then sending them was fine again, and now the problem reappeared. No change in bot's code during this time. 
Now I found what the problem was: Adaptive Card (v1.0) contained duplicate id's for elements, so I removed id's completely, and now it works fine.
